sdk/iothub/src/service/identity.rs

```diff
@@ -0,0 +1,363 @@
+use azure_core::errors::{extract_status_headers_and_body, AzureError, UnexpectedHTTPResult};
+
+use hyper::{Body, Client, Method, Request, StatusCode};
+use hyper_tls::HttpsConnector;
+
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+
+use crate::service::{ServiceClient, API_VERSION};
+
+/// Representation of a desired device capability
+pub enum DesiredCapability {
+    IotEdge,
+}
+
+/// The connection state of a module or device
+#[derive(Serialize, Debug, Deserialize, PartialEq)]
+pub enum ConnectionState {
+    Connected,
+    Disconnected,
+}
+
+/// Device or module status
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum Status {
+    Disabled,
+    Enabled,
+}
+
+/// Representation of device capabilities
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+pub struct DeviceCapabilities {
+    #[serde(rename = "iotEdge")]
+    pub iotedge: bool,
+}
+
+/// Representation of a symmetric key for authentication
+#[derive(Serialize, Deserialize, Debug, PartialEq, Default)]
+pub struct SymmetricKey {
+    primary_key: Option<String>,
+    secondary_key: Option<String>,
+}
+
+/// Representation of a x509 thumbprint for authentication
+#[derive(Default, Serialize, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct X509ThumbPrint {
+    pub primary_thumbprint: Option<String>,
+    pub secondary_thumbprint: Option<String>,
+}
+
+/// AuthenticationType of a module or device
+#[derive(Serialize, Debug, Deserialize, PartialEq)]
+pub enum AuthenticationType {
+    #[serde(rename = "certificate")]
+    Certificate,
```

Comment:
Nit: we should either add docs for each of the enum variants or `#[allow(missing_docs)]` for this enum. At some point we should turn on `!#[deny(missing_docs)]`.

Reply:
Thanks, I'll go over the rest of the `iothub` crate and add any missing docs and enable `#![deny(missing_docs)]` in a future PR.